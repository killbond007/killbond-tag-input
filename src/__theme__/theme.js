const theme = {
	breakpoints: ['768px', '1024px', '1240px'],
	devices: {
		sm: '@media (max-width: 576px)',
		md: '@media (max-width: 768px)',
		lg: '@media (max-width: 1024px)',
		xl: '@media (max-width: 1240px)',
	},
	colors: {
		purple_1: '#3827b0',
		purple_2: '#834cc4',
		pale_purple1: '#f9f9ff',
		red: '#ff7068',
		green: '#0cdcae',
		dark_indigo: '#180e5e',
		grey_shades: '#9f9bac',
		light_grey: '#c5d0de',
		dark: '#352c4b',
		white: '#ffffff',
	},
	opacities: {
		opacity_1: 0.85,
		opacity_2: 0.7,
		opacity_3: 0.55,
		opacity_4: 0.4,
		opacity_5: 0.25,
		opacity_6: 0.1,
	},
	fonts: {
		primary: 'Ubuntu',
		icons: 'Otto',
	},
	typography: {
		h1: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 24,
			fontWeight: 500,
			fontStyle: 'normal',
			lineHeight: 1.25,
			color: '#3827b0',
			'@media (max-width: 768px)': {
				fontSize: 22,
				lineHeight: 1.32,
			},
		},
		h2: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 18,
			fontWeight: 'normal',
			fontStyle: 'normal',
			lineHeight: 1.39,
			color: '#180e5e',
			'@media (max-width: 768px)': {
				fontSize: 17,
				lineHeight: 1.35,
			},
		},
		h3: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 16,
			fontWeight: 500,
			fontStyle: 'normal',
			lineHeight: 1.25,
			color: '#180e5e',
			'@media (max-width: 768px)': {
				fontSize: 15,
				lineHeight: 1.47,
			},
		},
		h4: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 15,
			fontWeight: 500,
			fontStyle: 'normal',
			lineHeight: 1.33,
			color: '#180e5e',
			'@media (max-width: 768px)': {
				fontSize: 14,
				lineHeight: 1.33,
			},
		},
		body: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 14,
			fontWeight: 'normal',
			fontStyle: 'normal',
			lineHeight: 1.43,
			color: '#352c4b',
			'@media (max-width: 768px)': {
				fontSize: 13,
				lineHeight: 1.54,
			},
		},
		body_bold: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 14,
			fontWeight: 700,
			fontStyle: 'normal',
			lineHeight: 1.43,
			color: '#352c4b',
			'@media (max-width: 768px)': {
				fontSize: 13,
				lineHeight: 1.54,
			},
		},
		label: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 12,
			fontWeight: 'normal',
			fontStyle: 'normal',
			lineHeight: 1.33,
			color: '#3827b0',
			'@media (max-width: 768px)': {
				fontSize: 11,
				lineHeight: 1.27,
			},
		},
		label_form: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 12,
			fontWeight: 'normal',
			fontStyle: 'normal',
			lineHeight: 1.33,
			color: '#9f9bac',
			'@media (max-width: 768px)': {
				fontSize: 11,
				lineHeight: 1.27,
			},
		},
		label_link: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 12,
			fontWeight: 'normal',
			fontStyle: 'normal',
			lineHeight: 1.33,
			color: '#9f9bac',
			'@media (max-width: 768px)': {
				fontSize: 11,
				lineHeight: 1.27,
			},
		},
		th: {
			fontFamily: 'Ubuntu, sans-serif',
			textTransform: 'uppercase',
			fontSize: 9,
			fontWeight: 'normal',
			fontStyle: 'normal',
			lineHeight: 1.33,
			color: '#9f9bac',
		},
		figure: {
			fontFamily: 'Ubuntu, sans-serif',
			fontSize: 40,
			fontWeight: 500,
			fontStyle: 'normal',
			lineHeight: 1.13,
			color: '#352c4b',
			'@media (max-width: 768px)': {
				fontSize: 30,
			},
		},
		icon_small: {
			'&&&': {
				fontFamily: 'Otto',
			},
			fontSize: 15,
			fontWeight: 'normal',
			fontStyle: 'normal',
			lineHeight: 1,
			'-webkit-font-smoothing': 'antialiased',
			'-moz-osx-font-smoothing': 'grayscale',
			color: '#ffffff',
		},
		icon_large: {
			'&&&': {
				fontFamily: 'Otto',
			},
			fontSize: 24,
			fontWeight: 'normal',
			fontStyle: 'normal',
			lineHeight: 1,
			'-webkit-font-smoothing': 'antialiased',
			'-moz-osx-font-smoothing': 'grayscale',
			color: '#ffffff',
		},
		icon_logo: {
			'&&&': {
				fontFamily: 'Otto',
			},
			fontSize: 35,
			fontWeight: 'normal',
			fontStyle: 'normal',
			lineHeight: 1,
			'-webkit-font-smoothing': 'antialiased',
			'-moz-osx-font-smoothing': 'grayscale',
			color: '#ffffff',
		},
	},
	shadows: {
		small: {
			boxShadow: '1px 1px 0 0 rgba(197, 208, 222, 0.4)',
		},
		medium: {
			boxShadow: '0 2px 1px 0 rgba(197, 208, 222, 0.7)',
		},
		large: {
			boxShadow: '0 0 2px 0 rgba(197, 208, 222, 0.7)',
		},
		table: {
			boxShadow: '4px 4px 4px 0 rgba(197, 208, 222, 0.4)',
		},
		focused: {
			boxShadow: '0 0 2px 0 #834cc4',
		},
	},
	borders: {
		light: {
			borderRadius: 3,
			border: '1px rgba(249, 249, 255, 1) solid',
		},
		regular: {
			borderRadius: 3,
			border: '1px rgba(197, 208, 222, 0.4) solid',
		},
		bold: {
			borderRadius: 3,
			border: '1px #c5d0de solid',
		},
		focused: {
			borderRadius: 3,
			border: '1px #834cc4 solid',
		},
		selected: {
			border: '4px #0cdcae solid',
		},
		error: {
			borderRadius: 3,
			border: '1px #ff7068 solid',
		},
	},
	lines: {
		separator: {},
	},
}

export default theme
