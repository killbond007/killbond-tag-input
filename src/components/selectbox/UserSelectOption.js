import React from 'react'
import PropTypes from 'prop-types'
import { noop } from 'lodash'

import * as Styled from './__styles__/UserSelectOption.styles'

const UserSelectOption = ({ value, label, onClick }) => {
	const _onClick = () => onClick(value, label)
	return (
		<Styled.Root data-testid="__user-select-option__" value={value} onClick={_onClick}>
			{label}
		</Styled.Root>
	)
}

UserSelectOption.propTypes = {
	/** Defines the value of the option */
	value: PropTypes.any,
	/** Defines the label of the option */
	label: PropTypes.string,
	/** Defines the handler called on click */
	onClick: PropTypes.func,
}

UserSelectOption.defaultProps = {
	value: null,
	label: null,
	onClick: noop,
}

export default UserSelectOption
