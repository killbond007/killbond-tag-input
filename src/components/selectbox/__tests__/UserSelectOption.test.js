import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import UserSelectOption from '../UserSelectOption'

import theme from '../../../__theme__/theme'

const defaultProps = {}
const getInstance = (props = {}) => (
	<ThemeProvider theme={theme}>
		<UserSelectOption {...defaultProps} {...props} />
	</ThemeProvider>
)

describe('UserSelectOption', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})

	it('renders label', () => {
		const label = 'Foo'
		const { getByText } = render(getInstance({ label }))
		expect(getByText(label)).toBeInTheDocument()
	})

	it('triggers onClick callback', () => {
		const onClick = jest.fn()
		const { getByTestId } = render(getInstance({ value: 'foo', label: 'Foo', onClick }))
		fireEvent.click(getByTestId('__user-select-option__'))
		expect(onClick).toHaveBeenCalledWith('foo', 'Foo')
	})
})
