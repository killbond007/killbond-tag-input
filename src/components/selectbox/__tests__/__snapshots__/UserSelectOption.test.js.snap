// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSelectOption matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  font-family: Ubuntu,sans-serif;
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  line-height: 1.43;
  color: rgb(24,14,94);
  min-height: 32px;
  padding-left: 8px;
  padding-right: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 3px;
}

.c0:hover {
  background-color: rgba(131,76,196,0.1);
}

@media (max-width:768px) {
  .c0 {
    font-size: 13px;
    line-height: 1.54;
  }
}

<option
    class="c0"
    data-testid="__user-select-option__"
  />
</DocumentFragment>
`;
